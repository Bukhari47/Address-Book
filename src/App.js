import "antd/dist/antd.css";
import React from "react";

import AppHeader from "./components/Header";
import HomePage from "./containers/HomePage";
import Settings from "./containers/Settings";
import User from "./containers/User";
import Page404Error from "./containers/Page404Error";

import { BrowserRouter as Router, Switch, Route } from "react-router-dom";

import { Layout } from "antd";
const { Header, Content, Footer } = Layout;

function App() {
  return (
    <Layout className="mainLayout">
      <Router>
        <Header>
          <AppHeader />
        </Header>
        <Content
          className="site-layout"
          style={{ padding: "0 50px", marginTop: 64 }}
        >
          <div
            className="site-layout-background"
            style={{ padding: 24, minHeight: 380 }}
          >
            <Switch>
              <Route path="/" exact component={HomePage} />
              <Route path="/settings" component={Settings} />
              <Route path="/user/:userId" component={User} />
              <Route component={Page404Error} />
            </Switch>
          </div>
        </Content>
      </Router>
      <Footer style={{ textAlign: "center" }}>
        Address book created by using React,Redux and ANTD
      </Footer>
    </Layout>
  );
}

export default App;
