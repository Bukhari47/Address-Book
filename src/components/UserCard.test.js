import React from "react";
import App from "../App";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

const mockStore = configureMockStore([thunk]);

describe("Users Card Should Be displayed", () => {
  it("Display user card", () => {
    const store = mockStore({
      startup: { complete: false },
    });
    const userCard = renderer.create(
      <Provider>
        <App store={store} />
      </Provider>
    );
    expect(userCard).toMatchSnapshot();
  });
});
