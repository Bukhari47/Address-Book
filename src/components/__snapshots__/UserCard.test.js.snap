// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users Card Should Be displayed Display user card 1`] = `null`;
